src/test/java/org/springframework/integration/aws/inbound/S3InboundChannelAdapterTests.java

```diff
@@ -150,6 +152,15 @@ void testS3InboundChannelAdapter() throws IOException {
 		assertThat(new File(LOCAL_FOLDER, "otherFile.a").exists()).isFalse();
 	}
 
+	@Test
+	void getHostPortWithEndpoint() {
+		AmazonS3 amazonS3 = Mockito.mock(AmazonS3.class);
+		S3SessionFactory s3SessionFactory = new S3SessionFactory(amazonS3);
+		s3SessionFactory.setEndpoint("s3-url.com:8000");
+
+		assertThat(s3SessionFactory.getSession().getHostPort()).isEqualTo("s3-url.com:8000");
```

Comment:
Well, my point was to inject your custom endpoint into an existing configuration.
See that `S3InboundFileSynchronizer` bean definition: one of its ctors is really based on the `S3SessionFactory`.
So, you can have an `S3SessionFactory` bean based on the existing `AmazonS3` bean, with your `endpoint`.
And in the test you would check a `FileHeaders.REMOTE_HOST_PORT` header from the polled message.

Reply:
Ok, I think I get it now